pennylane_qrack/qrack_device.py

```diff
@@ -199,36 +253,52 @@ def _apply_basis_state(self, op):
     def _apply_gate(self, op):
         """Apply native qrack gate"""
 
-        # translate op wire labels to consecutive wire labels used by the device
-        device_wires = self.map_wires(op.wires)
-        par = op.parameters
-
         opname = op.name
         if isinstance(op, Adjoint):
             op = op.base
             opname = op.name + ".inv"
 
-        if opname == "Toffoli" or opname == "Toffoli.inv" or opname == "CNOT" or opname == "CNOT.inv" or opname == "MultiControlledX" or opname == "MultiControlledX.inv":
+        if opname == "MultiRZ":
+            device_wires = self.map_wires(op.wires)
+            for q in device_wires:
+                self._state.r(Pauli.PauliZ, par[0], q)
+        elif opname == "C(MultiRZ)":
+            device_wires = self.map_wires(op.wires)
+            control_wires = self.map_wires(op.control_wires)
+            for q in device_wires:
+                self._state.mcr(Pauli.PauliZ, par[0], control_wires, q)
+
+        # translate op wire labels to consecutive wire labels used by the device
+        device_wires = self.map_wires((op.control_wires + op.wires) if op.control_wires else op.wires)
+        par = op.parameters
+
+        if opname == "Toffoli" or opname == "Toffoli.inv" or opname == "C(Toffoli)" or opname == "C(Toffoli).inv" or opname == "CNOT" or opname == "CNOT.inv" or opname == "C(CNOT)" or opname == "C(CNOT).inv" or opname == "MultiControlledX" or opname == "MultiControlledX.inv" or opname == "C(PauliX)" or opname == "C(PauliX).inv":
```

Comment:
You could make some of these statements a bit more concise by using the convention of checking list membership. For instance:

```py
if opname in ["Toffoli", "Toffoli.inv", "C(Toffoli)", ...]
```

The same would apply to the other statements as well.

Reply:
Thank you for the suggestion! Done.